data-pipeline/src/pipeline/pre_processing.py

```diff
@@ -249,7 +246,12 @@ def prepare_ks4_data(ks4_path):
         inplace=True,
     )
 
-    return ks4[["AverageAttainment", "Progress8Measure", "Progress8Banding"]].dropna()
+    ks4 = (
+        ks4[["URN", "AverageAttainment", "Progress8Measure", "Progress8Banding"]]
+        .dropna()
+        .drop_duplicates()
```

Comment:
Just as an FYI you need to be really careful with `drop_duplicates` it doesn't always result in what your expecting. This caused a nasty bug in CDC I spotted at the weekend. In general prefer `result[~result.index.duplicated(keep='first')]`

Reply:
That being said if you are sure this has the right behaviour feel free to resolve this comment.